libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2000,7 +2000,9 @@ void TypeChecker::endVisit(Literal const& _literal)
 			m_errorReporter.warning(
 				_literal.location(),
 				"This looks like an address but has an invalid checksum. "
-				"If this is not used as an address, please prepend '00'."
+				"If this is not used as an address, please prepend '00'. "
+				"Correct checksummed address: '" + _literal.getChecksumAddress() + "'. "
```

Comment:
`looksLikeAddress` cat retur true for strings of length 39, 40 and 41, but `getChecksumAddress` asserts that the string is exactly 40 hex digits long.

Reply:
Either we add another condition here and display the "correct" checksum if it is 40 characters or we just don't display it at all. It is worth displaying it?

Reply:
We should display the correct checksum for 40 characters. Otherwise, we tell the user the correct length. I think a compiler should always supply an "easy fix" method, if it exists. It is in the user's responsibility to accept it or not.

Reply:
What is the idea behind doing this for `41` hex digits?

Reply:
I basically see the following reasons for a checksum failure:

 - typing a wrong hex digit
 - swapping two hex digits
 - repeating a hex digit (or generally, adding a non-existing hex digit)
 - leaving out a hex digit

Because of that, we should treat hex digit sequences of lengths 39, 40 and 41 as possible addresses.